src/core/lib/iomgr/tcp_generic_options.h

```diff
@@ -0,0 +1,134 @@
+  int tcp_max_read_chunk_size;
+  int tcp_tx_zerocopy_send_bytes_threshold;
+  int tcp_tx_zerocopy_max_simultaneous_sends;
+  int tcp_tx_zero_copy_enabled;
+  int keep_alive_time_ms;
+  int keep_alive_timeout_ms;
+  int expand_wildcard_addrs;
+  int allow_reuse_port;
+  grpc_core::RefCountedPtr<grpc_core::ResourceQuota> resource_quota;
+  struct grpc_socket_mutator* socket_mutator;
+  TcpGenericOptions() : resource_quota(nullptr), socket_mutator(nullptr) {}
+  // Move ctor
+  TcpGenericOptions(TcpGenericOptions&& other) noexcept {
+    socket_mutator = absl::exchange(other.socket_mutator, nullptr);
+    resource_quota = std::move(other.resource_quota);
+    ResetAllIntegerOptions(other);
+  }
+  // Move assignment
+  TcpGenericOptions& operator=(TcpGenericOptions&& other) noexcept {
+    if (socket_mutator != nullptr) {
+      grpc_socket_mutator_unref(socket_mutator);
+    }
+    socket_mutator = absl::exchange(other.socket_mutator, nullptr);
+    resource_quota = std::move(other.resource_quota);
+    ResetAllIntegerOptions(other);
+    return *this;
+  }
+  // Copy ctor
+  TcpGenericOptions(const TcpGenericOptions& other) {
+    if (other.socket_mutator != nullptr) {
+      socket_mutator = grpc_socket_mutator_ref(other.socket_mutator);
+    } else {
+      socket_mutator = nullptr;
+    }
+    resource_quota = other.resource_quota;
+    ResetAllIntegerOptions(other);
+  }
+  // Copy assignment
+  TcpGenericOptions& operator=(const TcpGenericOptions& other) {
+    if (&other == this) {
+      return *this;
+    }
+    if (socket_mutator != nullptr) {
+      grpc_socket_mutator_unref(socket_mutator);
+      socket_mutator = nullptr;
+    }
+    if (other.socket_mutator != nullptr) {
+      socket_mutator = grpc_socket_mutator_ref(other.socket_mutator);
+    }
+    resource_quota = other.resource_quota;
+    ResetAllIntegerOptions(other);
+    return *this;
+  }
+  // Destructor.
+  ~TcpGenericOptions() {
+    if (socket_mutator != nullptr) {
+      grpc_socket_mutator_unref(socket_mutator);
+    }
+  }
+
+ private:
+  void ResetAllIntegerOptions(const TcpGenericOptions& other) {
+    tcp_read_chunk_size = other.tcp_read_chunk_size;
+    tcp_min_read_chunk_size = other.tcp_min_read_chunk_size;
+    tcp_max_read_chunk_size = other.tcp_max_read_chunk_size;
+    tcp_tx_zerocopy_send_bytes_threshold =
+        other.tcp_tx_zerocopy_send_bytes_threshold;
+    tcp_tx_zerocopy_max_simultaneous_sends =
+        other.tcp_tx_zerocopy_max_simultaneous_sends;
+    tcp_tx_zero_copy_enabled = other.tcp_tx_zero_copy_enabled;
+    keep_alive_time_ms = other.keep_alive_time_ms;
+    keep_alive_timeout_ms = other.keep_alive_timeout_ms;
+    expand_wildcard_addrs = other.expand_wildcard_addrs;
+    allow_reuse_port = other.allow_reuse_port;
+  }
+};
+
+// Initialize the tcp generic options struct members. This method can be used
+// to initialize dynamically allocated struct objects.
+void grpc_tcp_generic_options_init(TcpGenericOptions* options);
```

Comment:
Doesn't look like you addressed my earlier comment about this, but I don't think we need these init and destroy functions.  The options struct is a C++ type, and we should just instantiate and destroy it in the normal C++ ways.

Reply:
Removed them.